test/extensions/filters/network/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "test/extensions/filters/network/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::ReturnRef;
+using testing::WithArgs;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ExtAuthz {
+
+Filters::Common::ExtAuthz::ResponsePtr
+makeAuthzResponse(Filters::Common::ExtAuthz::CheckStatus status) {
+  Filters::Common::ExtAuthz::ResponsePtr response =
+      std::make_unique<Filters::Common::ExtAuthz::Response>();
+  response->status = status;
+  return response;
+}
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::network::ext_authz::ExtAuthzTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  }
+
+  Stats::TestUtil::TestStore stats_store;
+  Filters::Common::ExtAuthz::MockClient* client = new Filters::Common::ExtAuthz::MockClient();
+  envoy::extensions::filters::network::ext_authz::v3::ExtAuthz proto_config = input.config();
+
+  ConfigSharedPtr config = std::make_shared<Config>(proto_config, stats_store);
+  std::unique_ptr<Filter> filter =
+      std::make_unique<Filter>(config, Filters::Common::ExtAuthz::ClientPtr{client});
+
+  NiceMock<Network::MockReadFilterCallbacks> filter_callbacks;
+  filter->initializeReadFilterCallbacks(filter_callbacks);
+  Network::Address::InstanceConstSharedPtr addr =
+      std::make_shared<Network::Address::PipeInstance>("/test/test.sock");
+
+  ON_CALL(filter_callbacks.connection_, remoteAddress()).WillByDefault(ReturnRef(addr));
+  ON_CALL(filter_callbacks.connection_, localAddress()).WillByDefault(ReturnRef(addr));
+
+  for (const auto& action : input.actions()) {
+    switch (action.action_selector_case()) {
+    case envoy::extensions::filters::network::ext_authz::Action::kOnData: {
+      // Optional input field to set default authorization check result for the following "onData()"
+      if (action.on_data().has_result()) {
+        switch (action.on_data().result().result_selector_case()) {
+        case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusOk: {
+          ON_CALL(*client, check(_, _, _, _))
+              .WillByDefault(WithArgs<0>(
+                  Invoke([&](Filters::Common::ExtAuthz::RequestCallbacks& callbacks) -> void {
+                    callbacks.onComplete(
+                        makeAuthzResponse(Filters::Common::ExtAuthz::CheckStatus::OK));
+                  })));
+          break;
+        }
+        case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusError: {
+          ON_CALL(*client, check(_, _, _, _))
```

Comment:
drive-by: could all of these three cases be combined with something like
`callbacks.onComplete(makeAuthzResponse(resultToCheckStatus(action.on_data().result())))`

Reply:
> drive-by: could all of these three cases be combined with something like
> `callbacks.onComplete(makeAuthzResponse(resultToCheckStatus(action.on_data().result())))`

Good idea! Thanks. Updated.